wire/blockheader_test.go

```diff
@@ -156,6 +156,81 @@ func TestBlockHeaderWire(t *testing.T) {
 	}
 }
 
+// TestBlockHeaderBtcEncode tests BlockHeader encode and decode.
+func TestBlockHeaderBtcEncode(t *testing.T) {
```

Comment:
I'd prefer not having a separate test func here.  Just add the relevant encode/decode bits to `TestBlockHeaderWire`.  Also, the name doesn't match what the function does anyways since it's testing both encode/decode.

For example, after the call to

```Go
err := wire.TstWriteBlockHeader(&buf, test.pver, test.in)
...
if !bytes.Equal(buf.Bytes(), test.buf) {
...
```

Add:

```Go
+		err = test.in.BtcEncode(&buf, pver)
+		if err != nil { ... }
+		if !bytes.Equal(buf.Bytes(), test.buf) { ... }
```

Then the same deal for the decode.